src/main/java/com/opentok/OpenTok.java

```diff
@@ -490,6 +490,22 @@ public Archive stopArchive(String archiveId) throws OpenTokException {
     public void deleteArchive(String archiveId) throws OpenTokException {
         client.deleteArchive(archiveId);
     }
+
+    /**
+     * Patches an Opentok Archive
+     * <p>
+     * You can only patch an Archive that has a {@link com.opentok.Archive.StreamMode} of "manual"
+     * By patching an Opentok Archive you can add or remove a stream from archive.
+     *
+     * @param archiveId The archive ID of the archive you want to patch
+     * @param properties This {@link PatchProperties} object defining the patch body
+     *
+     * @throws OpenTokException
+     */
+    public void patchArchive(String archiveId, PatchProperties properties) throws OpenTokException {
+        client.patchArchive(archiveId, properties);
+    }
```

Comment:
I think we would be better off having two separate methods: `addArchiveSream(archiveId, streamId, hasAudio, hasVideo)` and  `removeArchiveStream(archiveId, streamId)`. This is because you cannot add and remove streams in the same API call (see https://github.com/opentok/api-review/blob/master/archiving-api/README.md?plain=1#L992-L1013).

The same comment applies for the broadcast-related method.

Have a look at the proposal for the Node SDK implementation: https://github.com/opentok/api-review/pull/232/.